test/test/smoke/SmokeTests.java

```diff
@@ -35,7 +35,7 @@ public void threads(TestProcess p) throws Exception {
 
     @Test(mainClass = LoadLibrary.class)
     public void loadLibrary(TestProcess p) throws Exception {
-        p.profile("-f %f -o collapsed -d 4 -i 1ms");
+        p.profile("-f %f -o collapsed -d 9 -i 1ms");
```

Comment:
This will extend running time by 5 seconds, which is a lot. I'd prefer not to do that.
What is the problem with the test?

Reply:
https://github.com/openorclose/async-profiler/actions/runs/10851672132/job/30115952174#step:2:162

```
INFO: Running SmokeTests.loadLibrary...
WARNING: A JVM TI agent has been loaded dynamically (/Users/runner/work/async-profiler/async-profiler/async-profiler/build/bin/../lib/libasyncProfiler.dylib)
WARNING: If a serviceability tool is in use, please run with -XX:+EnableDynamicAgentLoading to hide this warning
WARNING: Dynamic loading of agents will be disallowed by default in a future release
Profiling started
test/smoke/LoadLibrary.main;java/lang/Thread.sleep 1
test/smoke/LoadLibrary.main;java/lang/Thread.sleep 2
thread_start;_pthread_start;thread_native_entry;Thread::call_run;WatcherThread::run;WatcherThread::sleep;Monitor::wait_without_safepoint_check;PlatformMonitor::wait;__psynch_cvwait 3
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/lang/Thread.sleep0;JVM_Sleep;JavaThread::sleep_nanos;PlatformEvent::park_nanos;__gettimeofday 1
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/lang/Thread.sleep0;JVM_Sleep;JavaThread::sleep_nanos;PlatformEvent::park_nanos;__psynch_cvwait 7
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/lang/Thread.sleep0;JavaThread::is_interrupted 1
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/lang/Thread.sleep0;JVM_Sleep;JavaThread::sleep_nanos;PlatformEvent::park_nanos;_pthread_cond_wait;_pthread_mutex_droplock 1
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/util/concurrent/TimeUnit.toNanos 1
test/smoke/LoadLibrary.main;java/lang/Thread.sleep;java/lang/Thread.sleep0;JVM_Sleep;JavaThread::sleep_nanos;pthread_mutex_unlock 1
thread_start;_pthread_start;thread_native_entry;Thread::call_run;ConcurrentGCThread::run;G1ServiceThread::run_service;G1ServiceThread::run_task 1
WARNING: SmokeTests.loadLibrary failed
java.lang.AssertionError
	at test.smoke.SmokeTests.loadLibrary(SmokeTests.java:41)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at one.profiler.test.Runner.run(Runner.java:152)
	at one.profiler.test.Runner.run(Runner.java:[168](https://github.com/openorclose/async-profiler/actions/runs/10851672132/job/30115952174#step:2:169))
	at one.profiler.test.Runner.main(Runner.java:217)
```

`assert out.contains("Java_sun_management");` This assert fails

Reply:
Thanks. Probably, the following loop runs much longer than 2 seconds on macOS GitHub runner:
```
for (int i = 0; i < 200; i++) {
    Thread.sleep(10);
}
```
If this is the case, the loop needs to be adjusted (e.g., sleeping longer, but less frequently).
Increasing test duration is not the best solution.